dcp/api/job.py

```diff
@@ -167,14 +167,28 @@ def wait(self):
             return dry.aio.blockify(self._wait)()
 
         def on(self, *args):
+            # deserialize job on event parameters before passing them to user defined callback
+            def cb_deserialize_wrapper(callback):
+                def new_cb(*inner_args):
+                    nonlocal callback
+                    new_args = []
+                    for arg in inner_args:
+                        if isinstance(arg, dict) and bool(arg):
```

Comment:
For readability, would it be better 
```suggestion
                        if isinstance(arg, dict) and len(arg) > 0:
```

Reply:
honestly it's a no-op anyways, not sure why I check the length. It could just be

```python
if isinstance(arg, dict):
```

I'll change it to that

Reply:
changed in dfe1623